code/game/gamemodes/wizard/rightandwrong.dm

```diff
@@ -1,170 +1,169 @@
+	/obj/item/gun/projectile/automatic/l6_saw,
+	/obj/item/gun/projectile/automatic/m90,
+	/obj/item/gun/energy/alien,
+	/obj/item/gun/energy/gun/turret,
+	/obj/item/gun/energy/pulse/carbine,
+	/obj/item/gun/energy/decloner,
+	/obj/item/gun/energy/mindflayer,
+	/obj/item/gun/energy/kinetic_accelerator,
+	/obj/item/gun/energy/plasmacutter/adv,
+	/obj/item/gun/energy/wormhole_projector,
+	/obj/item/gun/projectile/automatic/wt550,
+	/obj/item/gun/projectile/automatic/shotgun/bulldog,
+	/obj/item/gun/projectile/revolver/grenadelauncher,
+	/obj/item/gun/projectile/revolver/golden,
+	/obj/item/gun/projectile/automatic/sniper_rifle,
+	/obj/item/gun/medbeam,
+	/obj/item/gun/energy/laser/scatter))
 
-	to_chat(usr, "<B>You summoned [summon_type ? "magic" : "guns"]!</B>")
-	message_admins("[key_name_admin(usr)] summoned [summon_type ? "magic" : "guns"]! ([revolver_fight ? "Revolver duel!" : ""] [fake_revolver_fight ? "Suicidal revolver duel!" : ""])")
+//if you add anything that isn't covered by the typepaths below, add it to summon_magic_objective_types
+GLOBAL_LIST_INIT(summoned_magic, list(
+	/obj/item/spellbook/oneuse/fireball,
+	/obj/item/spellbook/oneuse/smoke,
+	/obj/item/spellbook/oneuse/blind,
+	/obj/item/spellbook/oneuse/mindswap,
+	/obj/item/spellbook/oneuse/forcewall,
+	/obj/item/spellbook/oneuse/knock,
+	/obj/item/spellbook/oneuse/horsemask,
+	/obj/item/spellbook/oneuse/charge,
+	/obj/item/spellbook/oneuse/summonitem,
+	/obj/item/gun/magic/wand,
+	/obj/item/gun/magic/wand/death,
+	/obj/item/gun/magic/wand/resurrection,
+	/obj/item/gun/magic/wand/polymorph,
+	/obj/item/gun/magic/wand/teleport,
+	/obj/item/gun/magic/wand/door,
+	/obj/item/gun/magic/wand/fireball,
+	/obj/item/gun/magic/staff/healing,
+	/obj/item/gun/magic/staff/door,
+	/obj/item/scrying,
+	/obj/item/voodoo,
+	/obj/item/clothing/suit/space/hardsuit/wizard,
+	/obj/item/immortality_talisman,
+	/obj/item/melee/ghost_sword))
+
+GLOBAL_LIST_INIT(summoned_special_magic, list(
+	/obj/item/gun/magic/staff/change,
+	/obj/item/gun/magic/staff/animate,
+	/obj/item/storage/belt/wands/full,
+	/obj/item/contract,
+	/obj/item/gun/magic/staff/chaos,
+	/obj/item/necromantic_stone,
+	/obj/item/blood_contract))
+
+//everything above except for single use spellbooks, because they are counted separately (and are for basic bitches anyways)
+GLOBAL_LIST_INIT(summoned_magic_objectives, list(
+	/obj/item/contract,
+	/obj/item/blood_contract,
+	/obj/item/clothing/suit/space/hardsuit/wizard,
+	/obj/item/gun/magic,
+	/obj/item/immortality_talisman,
+	/obj/item/melee/ghost_sword,
+	/obj/item/necromantic_stone,
+	/obj/item/scrying,
+	/obj/item/spellbook,
+	/obj/item/storage/belt/wands/full,
+	/obj/item/voodoo))
+
+// If true, it's the probability of triggering "survivor" antag.
+GLOBAL_VAR_INIT(summon_guns_triggered, FALSE)
+GLOBAL_VAR_INIT(summon_magic_triggered, FALSE)
+
+/proc/give_guns(mob/living/carbon/human/H)
+	if(H.stat == DEAD || !(H.client))
+		return
+	if(H.mind)
+		if(iswizard(H))
```

Comment:
So, a difference from TG: they have it blocked so that if you have the gun survivalist objective, you can't get it again.

Quite frankly, if a wizard wants to spend most of his spell points on just summoning tons of guns and getting the crew to slaughter each other...well, so be it; likewise, it makes it more fun for admins to spam the spell for events and such.

It also is bugged on TG; if summon guns/magic first, then summon the opposite, multiple times, the second one will bypass the limit, since the check is just for all types of `/datum/antag`.